exercises/linked-list/src/example.h

```diff
@@ -0,0 +1,34 @@
+#ifndef LINKED_LIST_H
+#define LINKED_LIST_H
+
+#include <stdbool.h>
+
+typedef int ll_data_t;
+
+struct list_item;
+
+// constructs new item holding specified data
+struct list_item *new(ll_data_t);
+
+// retrieves data from item
+ll_data_t item_data(struct list_item *item);
+
+// inserts item at back list
+bool push(struct list_item *item);
```

Comment:
It looks like this is a singleton. I haven't yet decided whether this makes sense for simplicity or whether this is a bit odd. Can you give your reasoning?

Reply:
Sorry I was referring to the module overall.